Xamarin.Forms.Platform.Android/Renderers/NavigationRenderer.cs

```diff
@@ -39,17 +40,24 @@ public Task<bool> PushViewAsync(Page page, bool animated = true)
 
 		protected override void Dispose(bool disposing)
 		{
-			if (disposing)
+			if (disposing && !_disposed)
 			{
-				foreach (VisualElement child in PageController.InternalChildren)
-				{
-					IVisualElementRenderer renderer = Platform.GetRenderer(child);
-					if (renderer != null)
-						renderer.Dispose();
-				}
+				_disposed = true;
 
 				if (Element != null)
 				{
+					foreach (Element element in PageController.InternalChildren)
```

Comment:
Just verifying that there's no chance Element could be null yet we haven't cleaned all this up yet, since we're moving this inside the `if` statement.